src/clj/rems/db/entitlements.clj

```diff
@@ -50,7 +50,8 @@
   {:ga4gh_visa_v1 (mapv entitlement-to-permissions-api
                         (db/get-entitlements {:user            (if (has-roles? :handler :owner :organization-owner :reporter)
                                                                  user-or-nil
-                                                                 (getx-user-id))
+                                                                 (when-not (= user-or-nil (getx-user-id))
```

Comment:
now the `:user` parameter to `db/get-entitlements` will be nil for applicants. perhaps you could just have the throw-unauthorized check before the db call?

also, I think user-or-nil can no longer be nil